client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( action.area === state.current || ! isValidArea( action.area ) ) {
+				return state;
+			}
+			return Object.assign( {}, state, { current: action.area, previous: state.current } );
+		case LAYOUT_NEXT_FOCUS_SET:
+			if ( action.area === state.next || ! isValidArea( action.area ) ) {
+				return state;
+			}
+			return Object.assign( {}, state, { next: action.area } );
+		case LAYOUT_NEXT_FOCUS_ACTIVATE:
+			// If we don't have a change queued and the focus has changed
+			// previously, set it to `content`. This avoids having to set the
+			// focus to content on all navigation links because it becomes the
+			// default after focus has shifted.
+			if ( ! state.next && ! state.previous ) {
```

Comment:
Ok, I think I better understand the intent here:

On every page load, we automatically call `activateNextLayoutFocus`. If something has called `setNextLayoutFocus`, then shift `current` to that. However, if no there is no `next` value set, *and* this page load is originating from a sidebar or site picker, change `current` to `'content'`. This way, links which originate from the sidebar or site picker automatically hide those elements without having to do so explicitly.

`previous` is used to determine the part above where we need to know if the page load is originating from a sidebar or site picker.

That said, if the page load is *not* originating from the sidebar or site picker (or preview), it's likely originating from `content`, so there's really no danger in setting the current state to `content` in that case (or more precisely, setting the current state to `content` if there is no `next` and `content` is not the current state).

I'll give that a try.